### PR TITLE
chore: rename `acvm-simulator` to `acvm_js`

### LINKED ISSUE #54: Bikeshed the naming of this package. ("ACVM simulator isn't a simulator")
This package has been called ACVM simulator since its inception however as mentioned by @kobyhallx, it's not really a fitting name.

ACVM simulator doesn't "simulate" the ACVM in any real sense more than nargo does; it's just a javascript wrapper around the code necessary to _execute_ the ACVM. Before we make the initial release we should then explore alternative names for it.

Raised possibilities include:
- acvm-executor (technically true but seems overly verbose to me, c.f [revm](https://github.com/bluealloy/revm))
- acvm
- acvm.js

### PR DESCRIPTION
**Pull request by TomAFrench**
# Description

## Problem\*

Resolves #54 

## Summary\*

This PR renames `acvm-simulator` to `acvm-js`

## Additional Context

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by TomAFrench** [comment]
Asking for review for a sign-off on the name.

File: buildPhaseCargoCommand.sh [hunk 1]
```diff
@@ -5,9 +5,9 @@ if [ -d ./pkg/ ]; then
     rm -rf ./pkg/
 fi
 
-WASM_BINARY=./target/wasm32-unknown-unknown/release/acvm_simulator.wasm
-NODE_WASM=./pkg/nodejs/acvm_simulator_bg.wasm
-BROWSER_WASM=./pkg/nodejs/acvm_simulator_bg.wasm
+WASM_BINARY=./target/wasm32-unknown-unknown/release/acvm_js.wasm
+NODE_WASM=./pkg/nodejs/acvm_js_bg.wasm
+BROWSER_WASM=./pkg/nodejs/acvm_js_bg.wasm
```
Comment (kobyhallx) [inline buildPhaseCargoCommand.sh:10]: should we use pname pattern?

File: flake.nix [hunk 2]
```diff
@@ -95,7 +95,7 @@
       GIT_DIRTY = if (self ? rev) then "false" else "true";
 
       commonArgs = {
-        pname = "acvm-simulator";
+        pname = "acvm-js";
```
Comment (kobyhallx) [inline flake.nix:98]: This name should be adopted everywhere same to be reused `-` vs `_`

**Comment by kobyhallx** [comment]
Manual `./build.sh` fails with 
```
./buildPhaseCargoCommand.sh: line 16: wasm-opt: command not found
./buildPhaseCargoCommand.sh: line 17: wasm-opt: command not found
```

**Pull request merged**